metals/src/main/scala/scala/meta/internal/metals/ModuleStatus.scala

```diff
@@ -0,0 +1,192 @@
+package scala.meta.internal.metals
+
+import java.util.concurrent.ConcurrentHashMap
+
+import scala.concurrent.Promise
+
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.clients.language.MetalsLanguageClient
+import scala.meta.internal.metals.clients.language.MetalsStatusParams
+import scala.meta.internal.metals.clients.language.StatusType
+import scala.meta.internal.tvp.MetalsCommand
+import scala.meta.io.AbsolutePath
+import scala.meta.pc.reports.Report
+
+import ch.epfl.scala.bsp4j.BuildTargetIdentifier
+
+class ModuleStatus(
+    client: MetalsLanguageClient,
+    focusedPath: () => Option[AbsolutePath],
+    serviceForPath: AbsolutePath => {
+      def buildTargets: BuildTargets
```

Comment:
Could we have trait for this?

Reply:
We can, but then `MetalsLspServer` has to extend this trait and it is not used anywhere else. I can also just substitute this with `MetalsLspServer`, but as is, it feels more explicit, which parts are used.

Reply:
I am jsut wondering it it's not more costly in runtime, I remember someone somewhere saying that this is basically doing reflection underneath. Though I might be misremembering things

Reply:
I can't find any indication that it might be a problem though, so maybe ignore me